theano/sandbox/cuda/basic_ops.py

```diff
@@ -3329,20 +3329,42 @@ def c_code_cache_version(self):
         return ()
 
 
-class GpuFlatten(gof.HideC, tensor.Flatten, GpuOp):
+class GpuFlatten(gof.HideC, tensor.Reshape, GpuOp):
```

Comment:
Leave ```Flatten``` here. We want ```GpuFlatten``` and ```Flatten``` to stay bounded together. Actually, I would leave ```Flatten.make_node``` ```GpuFlatten.make_node``` just to make it clear. Sorry for changing my mind so often! ^_^